data/moves.js

```diff
@@ -945,8 +945,11 @@ let BattleMovedex = {
 		accuracy: 100,
 		basePower: 60,
 		basePowerCallback: function (pokemon, target, move) {
-			if (target.lastDamage > 0 && pokemon.lastAttackedBy && pokemon.lastAttackedBy.thisTurn && pokemon.lastAttackedBy.pokemon === target) {
-				this.debug('Boosted for getting hit by ' + pokemon.lastAttackedBy.move);
+			let hurtByTarget = pokemon.hurtBy.find(function (x) {
+				return x.source === target && x.damage > 0 && x.thisTurn;
+			});
```

Comment:
You don't seem to use the found value. Did you mean `some` rather than `find`?

Also, your code style prefers new-style lambdas here:

```
let hurtByTarget = pokemon.hurtBy.some(h =>
  p.source === target && p.damage > 0 && p.thisTurn;
);
```

Reply:
I use the found value in the debug output to get the move that triggered the boost. If that's not too important I can use `some` and a more generic debug statement.